File: LOTS/Projet/lots-java-template/Templates/src/graphe/Arete.java
package graphe;
import java.util.ArrayList;


public class Arete <G extends Graphe<G,S,A>, S extends Sommet<G,S,A>, A extends Arete<G,S,A>>{

	protected String nomArete;
	protected G Graphe;
	protected ArrayList<S> sommets;
	public static int num = 0;
	
	public Arete(){
		num++;
		this.nomArete = "Arete" + num;
		this.Graphe = null;
		this.sommets = new ArrayList<S>(2);
	}
	
	public String getNom(){
		return this.nomArete;
	}
	
	public String getNomSommetsLies(){
		return sommets.get(0).getNom() + " " + sommets.get(1).getNom() + " ";
	}
	
	
	public Arete(String n){
		this.nomArete = n;
		this.Graphe = null;
		this.sommets = new ArrayList<S>(2);
	}
	
	public void setSommets(S s1, S s2){
		this.sommets.add(s1);
		this.sommets.add(s2);
	}
	
	public ArrayList<S> getListeSommet(){
		return this.sommets;
	}
	
	public void setGraphe(G g){
		this.Graphe = g;
	}
}

File: LOTS/Projet/scala/Scala-lots/src/virtuel/Arete.scala
package virtuel

object numA{
  var current = 0
  def inc = {current += 1; current}
}

abstract class Arete {
  
  type virtualSommet <: Sommet;
  type virtualArete <: Arete;
  type virtualGraphe <:Graphe;
  
  var graphe: virtualGraphe = _;
  var nomArete : String =  "Arete" + numA.inc;
  var sommet1 : virtualSommet =_;
  var sommet2 : virtualSommet = _;
  
  def getNom : String = {
    return this.nomArete
  }
  
  def getSommet1 : virtualSommet = {
    return this.sommet1
  }
  
  def getSommet2 : virtualSommet = {
    return this.sommet2
  }
  
  def setSommets(s1 : virtualSommet, s2 : virtualSommet){
    this.sommet1 = s1;
    this.sommet2 = s2;
  }
  
  def setGraphe(g : virtualGraphe){
    this.graphe = g
  }
  
  def getNomSommetsLies: String = {
    return "(" + sommet1.getNom + " -> " + sommet2.getNom + ") "
  }
  
}